QtCollider/widgets/QcScopeShm.cpp

```diff
@@ -247,15 +252,18 @@ void QcScopeShm::paint1D( bool overlapped, int chanCount, int maxFrames, int fra
       painter.save();
       painter.translate( area.x(), area.y() + yOrigin );
       painter.setPen(pen);
+      
+      qreal ratio = 1.0 / _pixmap.devicePixelRatio();
 
       QPainterPath pathLine;
       QPainterPath pathFill;
 
-      int p=0, f=1; // pixel, frame
+      qreal p=0;
+      int f=1; // pixel, frame
       float min, max;
       min = max = frameData[0];
 
-      while( p++ < w )
```

Comment:
thinking about this more, this is a bug in your change - in the previous version, `p++ < w` would execute an iteration for `p == w`. now, the loop will terminate before p == w. do you want that?

Reply:
oops, must've typed a response and not sent. you are correct here - this is a bug, should be the for loop you mentioned elsewhere.

Reply:
ok! i'll make that change tomorrow and then we should be all set :)

Reply:
ok! i'll make that change tomorrow and then we should be all set :)

Reply:
ok! i'll make that change tomorrow and then we should be all set :)

Reply:
8 days later...